Checkpoint/connectors/checkpoint_harmony.py

```diff
@@ -0,0 +1,153 @@
+
+        # If undefined, retrieve events from the last 1 hour
+        if last_event_date_str is None:
+            return one_hour_ago
+
+        # Parse the most recent date seen
+        last_event_date = isoparse(last_event_date_str).replace(microsecond=0)
+
+        # We don't retrieve messages older than 1 hour
+        if last_event_date < one_hour_ago:
+            return one_hour_ago
+
+        return last_event_date
+
+    def get_checkpoint_client(self) -> CheckpointHttpClient:
+        """Get CheckpointHttpClient."""
+        if self._checkpoint_client is None:
+            self._checkpoint_client = CheckpointHttpClient(
+                client_id=self.module.configuration.client_id,
+                secret_key=self.module.configuration.secret_key,
+                auth_url=self.module.configuration.authentication_url,
+                base_url=self.module.configuration.base_url,
+            )
+
+        return self._checkpoint_client
+
+    async def get_checkpoint_harmony_events(self) -> list[str]:
+        """
+        Get CheckpointHarmony events.
+
+        Returns:
+            list[str]:
+        """
+        _last_event_date = self.last_event_date
+        events = await self.get_checkpoint_client().get_harmony_mobile_alerts(_last_event_date, 100)
+
+        logger.info("Fetched {0} events from source", len(events))
+
+        _new_latest_event_date = _last_event_date
+        for event in events:
+            # Swagger says that it should have "%m/%d/%Y %H:%M:%S" but all other values that we pass are in ISO format
+            # So we try to parse it as ISO and if it fails we parse it as "%m/%d/%Y %H:%M:%S"
+            # See 200 response here:
+            # https://app.swaggerhub.com/apis-docs/Check-Point/harmony-mobile/1.0.0-oas3#/Events/GetAlerts
+            try:
+                event_date = isoparse(event.event_timestamp)
+            except ValueError:
+                event_date = datetime.strptime(event.event_timestamp, "%m/%d/%Y %H:%M:%S")
+
+            if event_date > _new_latest_event_date:
+                _new_latest_event_date = event_date
+
+        result: list[str] = await self.push_data_to_intakes(
+            [orjson.dumps(event.dict()).decode("utf-8") for event in events]
+        )
+
+        with self.context as cache:
+            cache["last_event_date"] = _new_latest_event_date.isoformat()
+
+        return result
+
+    def run(self) -> None:  # pragma: no cover
+        """Runs Crowdstrike Telemetry."""
+        previous_processing_end = None
+
+        while self.running:
+            try:
+                loop = asyncio.get_event_loop()
+
+                while self.running:
+                    processing_start = time.time()
+                    if previous_processing_end is not None:
+                        EVENTS_LAG.labels(intake_key=self.configuration.intake_key).observe(
+                            processing_start - previous_processing_end
+                        )
```

Comment:
Can you use the `latest_event_date` to compute the lag? Like in the [cybereason connector](https://github.com/SEKOIA-IO/automation-library/blob/670123d1651c506afbf8ae8ef6a214c44aa33151/Cybereason/cybereason_modules/connector_pull_events.py#L316-L322)?